src/seth/libexec/seth/seth-4byte-decode

```diff
@@ -0,0 +1,21 @@
+#!/usr/bin/env bash
+### seth-4byte-decode -- querys 4byte.directory to find a matching signature, then decodes the calldata with it
+### Usage: seth 4byte-decode <calldata>
+###
+### Querys 4byte.directory to find matching signatures and prompts the user to to select one. The selected
+### signature will be used to decode the calldata.
```

Comment:
could be nice to have an `--optimistic` (I feel lucky) flag to this if we wanna build some larger pipeline where we can't be interactive

Reply:
I was thinking that too, just wasn't sure of how we'd want to do it. You can sometimes deduce which sig is the correct one from the calldata, but that can get hairy for complex sigs, and you can't always deduce it

For now I learn towards a simpler approach of having `--optimistic` use the oldest sig 4byte.directory has. Anecdotally this seems to often be the desired one (or at least it was the for the ERC20 methods I used for testing)

Reply:
wdyt about a `--id` flag that accepts either an id number or `earliest`/`latest`

Reply:
Hmm, what would be the use case for specifying an ID? If you know the ID then presumably you also know the function signature, meaning you could use `seth --calldata-decode` instead of `seth 4byte-decode`

Reply:
meh mainly for the ability to specify `earliest`/`latest`

and id is shorter than using the full sig 🤷

Reply:
That seems reasonable enough to me, being able to toggle earliest/latest is nice

Reply:
Would love some help with my `--id` flag here. In https://github.com/dapphub/dapptools/pull/757/commits/f45ee07deda6ec72da72945338fb1233b9705646 I pushed an attempt that works if I set the env var manually but not with the `--id` flag